apps/src/code-studio/pd/foorm/FoormLibraryEditorManager.jsx

```diff
@@ -23,81 +25,125 @@ const styles = {
+    availableLibraryQuestionsForCurrentLibrary: PropTypes.array,
+    resetAvailableLibraryQuestions: PropTypes.func,
     setLastSaved: PropTypes.func,
     setSaveError: PropTypes.func,
-    setFormData: PropTypes.func,
+    setLibraryQuestionData: PropTypes.func,
     setHasError: PropTypes.func,
-    setLastSavedQuestions: PropTypes.func
+    setLastSavedQuestion: PropTypes.func,
+    setLibraryData: PropTypes.func
   };
 
   constructor(props) {
     super(props);
 
     this.state = {
-      formKey: 0,
-      formPreviewQuestions: null,
       showCodeMirror: false,
       hasLoadError: false
     };
 
-    this.props.resetAvailableForms(this.props.formNamesAndVersions);
-  }
-
-  componentDidUpdate(prevProps) {
-    if (prevProps.formNamesAndVersions !== this.props.formNamesAndVersions) {
-      this.props.resetAvailableForms(this.props.formNamesAndVersions);
-    }
+    this.props.resetAvailableLibraries(this.props.libraryNamesAndVersions);
   }
 
+  // could probably dedupe this method
   getFormattedConfigurationDropdownOptions() {
-    return this.props.availableForms.map((formNameAndVersion, i) => {
-      const formName = formNameAndVersion['name'];
-      const formVersion = formNameAndVersion['version'];
-      const formId = formNameAndVersion['id'];
-      return (
-        <MenuItem
-          key={i}
-          eventKey={i}
-          onClick={() => this.loadConfiguration(formId)}
-        >
-          {`${formName}, version ${formVersion}`}
-        </MenuItem>
+    return this.props.availableLibraries.map((libraryNameAndVersion, i) => {
+      const libraryName = libraryNameAndVersion['name'];
+      const libraryVersion = libraryNameAndVersion['version'];
+
+      return this.getMenuItem(
+        () => this.loadLibraryQuestions(libraryNameAndVersion),
+        `${libraryName}, version ${libraryVersion}`,
+        i
       );
     });
   }
 
-  loadConfiguration(formId) {
+  getFormattedLibraryQuestionDropdownOptions() {
+    return this.props.availableLibraryQuestionsForCurrentLibrary.map(
+      (libraryQuestionAndType, i) => {
+        const libraryQuestionName = libraryQuestionAndType['name'];
+        const libraryQuestionType = libraryQuestionAndType['type'];
+        const libraryQuestionId = libraryQuestionAndType['id'];
+
+        return this.getMenuItem(
+          () => this.loadLibraryQuestionContent(libraryQuestionId),
+          `${libraryQuestionName} (${libraryQuestionType})`,
+          i
+        );
+      }
+    );
+  }
+
+  getMenuItem = (clickHandler, textToDisplay, key) => {
+    return (
+      <MenuItem key={key} eventKey={key} onClick={clickHandler}>
+        {textToDisplay}
+      </MenuItem>
+    );
+  };
+
+  loadLibraryQuestions(selectedLibraryNameAndVersion) {
+    let libraryId = selectedLibraryNameAndVersion['id'];
+
     this.props.setLastSaved(null);
     this.props.setSaveError(null);
+    this.props.setLibraryData(selectedLibraryNameAndVersion);
+    this.updateLibraryQuestionData({
```

Comment:
What's the purpose of `setLastSaved(null)`, `setSaveError(null)` and `this.updateLibraryQuestionData` here? to clear out some state before loading new data?

Reply:
The `this.updateLibraryQuestionData` is there (like you said) to wipe the selected question if a user selects a new library after having already selected a library question in another library. 

Totally makes sense these should be grouped / made more clear what they're doing, let me see if I can clean that up.